trie/utils/verkle.go

```diff
@@ -262,16 +267,25 @@ func EvaluateAddressPoint(address []byte) *verkle.Point {
 }
 
 func GetTreeKeyStorageSlotWithEvaluatedAddress(evaluated *verkle.Point, storageKey *uint256.Int) []byte {
-	pos := storageKey.Clone()
+	// Note that `pos` must be a big.Int and not a uint256.Int, because the subsequent
+	// arithmetics operations could overflow. (e.g: imagine if storageKey is 2^256-1)
+	pos := storageKey.ToBig()
```

Comment:
the caller of this function always creates a uint256 from a hash, for the sole purpose of calling this function, so the signature of this function should either:
  1. be changed to accept a `big.Int`
  2. accept `[]byte` directly and perform the math directly without using `big.Int`, or create the `big.Int` inside the function as an implementation detail, to be optimized later.

I think option 2 is the best approach, using a `big.Int` for now, so that we don't introduce a bug reinventing the wheel.

Reply:
Agree, when with option 2 receiving a `[]byte` and `SetBytes` to `big.Int`; changed some calling sites if you want to double check.